aiohttp/web_request.py

```diff
@@ -510,10 +511,42 @@ def text(self):
         return bytes_body.decode(encoding)
 
     @asyncio.coroutine
-    def json(self, *, loads=json.loads):
-        """Return BODY as JSON."""
+    def json(self, *, loads=json.loads, allowed_types=None):
+        """ Return BODY as JSON. If the `Content-Type` is not in allowed_types,
+        then the request is rejected.
+        """
+        content_type = self.content_type
+        (mime_type, subtype, suffix, parameters) = parse_mimetype(content_type)
+        content_type = mime_type + '/' + subtype
+        if suffix != '':
+            content_type += '+' + suffix
+
+        charset = parameters.get('charset', None)
+
+        if isinstance(allowed_types, str):
+            if allowed_types == 'json':
```

Comment:
This means 'json-like'. View my code after that. I will then check  whether the `Content-Type` is 'application/json' or starts with 'application' and ends with '+json'. If it is a string and is not 'json', I will compare it with `Content-Type` directly.